### ISSUE TITLE
Broken normalization layer save and load: very different results when using a model after it has been dumped to disk and reloaded if it uses normalization layers

### ISSUE THREAD

**Issue by jerabaul29** [keras-team/keras#17486]
**System information**.

- No custom code written by myself (except for preparing the data of course)
- Ubuntu 20.04
- TensorFlow installed with pip3
- TensorFlow version: v2.11.0-rc2-17-gd5b57ca93e5 2.11.0
- keras version: pip installed, 2.11.0
- Python version: Python 3.8.10
- Bazel version (if compiling from source): not applicable (installed with pip3)
- GPU model and memory: not relevant
- Exact command to reproduce: see below

**Describe the problem**.

I get very different predictions on the same data, before and after a cycle of model dump / del / load.

The model looks like:

```python
labels_inv_normalization_layer = layers.Normalization(invert=True, input_shape=[1,], axis=None)
labels_inv_normalization_layer.adapt(np_labels_training)

predictors_normalization_layer = layers.Normalization()
predictors_normalization_layer.adapt(np_predictors_training)

input_layer = keras.layers.Input(shape=(np_predictors_training.shape[1], ))
normalized_input = predictors_normalization_layer(input_layer)
fully_connected_1 = keras.layers.Dense(60, activation="relu")(normalized_input)
fully_connected_2 = keras.layers.Dense(60, activation="relu")(fully_connected_1)
fully_connected_3 = keras.layers.Dense(60, activation="relu")(fully_connected_2)
fully_connected_4 = keras.layers.Dense(60, activation="relu")(fully_connected_3)
output = keras.layers.Dense(1)(fully_connected_4)
denormalized_output = labels_inv_normalization_layer(output)

# model
keras_model = keras.Model(inputs=input_layer, outputs=denormalized_output)
keras_model.compile(optimizer=keras.optimizers.Adagrad(learning_rate=learning_rate), loss=loss_kind)
```

I perform prediction with:

```python
nn_predicted_training = np.array(keras_model(np_predictors_training)).squeeze()
nn_predicted_validation = np.array(keras_model(np_predictors_validation)).squeeze()
```

I dump, clear, and re-load the model with:

```python
keras_model.save("./saved_models/keras_model_vanilla_ml_ipynb3")
del keras_model
keras_model = load_model("./saved_models/keras_model_vanilla_ml_ipynb3")
```

**Describe the current behavior**.

After training the model, before dumping, I get a set of predictions on some validation data that I am happy with. After dumping, del, and loading the model, and using the loaded model on the exact same validation data, I get a very different output prediction (that are completely off).

I get a warning message when dumping:

```
WARNING:absl:Found untraced functions such as _update_step_xla while saving (showing 1 of 1). These functions will not be directly callable after loading.
INFO:tensorflow:Assets written to: ./saved_models/keras_model_vanilla_ml_ipynb3/assets
INFO:tensorflow:Assets written to: ./saved_models/keras_model_vanilla_ml_ipynb3/assets
```

Though it looked like checking on the issue tracker here that it should not be an issue?

**Describe the expected behavior**

I would expect to get the exact same predictions on the same validation data before model saving and after model del and loading.

**Standalone code to reproduce the issue**.

The notebook I use is private for now and not minimal. If there are doubts about this bug report I can try to set up a minimal example, but for now I would like to check if there is an obvious answer from this report alone :) .

**Comment by jerabaul29** [comment]
I was able to set up a least complexity example and to identify that the normalization layer(s) are well the cause of the issue. Without the normalization layer, things work well (see this notebook):

https://github.com/jerabaul29/misc/blob/main/BugReports/LeastExampleKerasBugNetworkDumpWithNormalization/examle_work_without_normalization.ipynb

But with normalization layer, the restored network is way off (likely, some normalization not or wrongly applied):

https://github.com/jerabaul29/misc/blob/main/BugReports/LeastExampleKerasBugNetworkDumpWithNormalization/exmaple_fail_with_normalization.ipynb

**Comment by jerabaul29** [comment]
My feeling, but I may well be wrong as I am not too familiar with the internals of the Keras code base, is that this:

https://github.com/keras-team/keras/blob/8331e3b0c8253d85c793658917dc312d5a4e87bc/keras/layers/preprocessing/normalization.py#L371-L380

misses the ```ìnvert``` attribute. This would explain why the test:

https://github.com/keras-team/keras/blob/8331e3b0c8253d85c793658917dc312d5a4e87bc/keras/layers/preprocessing/normalization_test.py#L428-L457

does not catch this issue: there is no test agains a case with the ```invert``` attribute set; that may mean that in addition to the code fix, an additional test case ```test_saved_model_tf_invert``` would be useful.

**Comment by jerabaul29** [comment]
I think this hypothesis is confirmed: when removing the ```invert``` layer and keeping the non ```invert``` layer, saving and loading works just fine:

https://github.com/jerabaul29/misc/blob/main/BugReports/LeastExampleKerasBugNetworkDumpWithNormalization/exmaple_fail_with_normalization_ok_noinvert.ipynb

**Comment by jerabaul29** [comment]
Confirmed: I added:

```
                "invert": self.invert,
```

to the ```get_config``` and this fixes the bug. I will now prepare a pull request with this addition and an additional test for ```invert``` layers.

**Comment by sushreebarsa** [comment]
@jerabaul29 Thank you for the update!

**Comment by jerabaul29** [comment]
Thanks for the reminder that the pull request got merged, sorry I had forgotten to close this report issue, closing it now :) .
